chidioguejiofor/volunteersng-api
# RUNO: Migrate Sequelize to Typeorm

#### What does this PR achieve?
The PR migrates the project from sequelize to Typeorm and it consists of the following actions:
- Installs typeorm, typeorm/nest and pg
- Imports ormconfig into app module and adds it to imports array
- writes entities for user, role, membership and organization
- generates migrations file for all entities
- moves the test folder back to the root of the project as it was before i moved it to the src directory. I did this because it was causing some hidden errors.
#### Any background context?
#### Manual Testing Steps
- `npm install`
- Go to `.env` or create one (follow .env-sample) and add your database the credentials 
- run `npm run typeorm:run` and check ormlogs.log and also open database to see tables created
- run `npm run typeorm:revert` to drop all the tables in the database
#### Screenshots:
#### Any new deployment instructions
#### New environmental Variables?
In `.env`: 
```
# Developement Database
DEV_PG_PORT=<number>
DEV_PG_HOST=<string>
DEV_PG_USER=<string>
DEV_PG_PASSWORD=<string>
DEV_PG_DATABASE=<string>
# Test Database
TEST_PG_HOST=<string>
TEST_PG_PORT=<number>
TEST_PG_USER=<string>
TEST_PG_PASSWORD=<string>
TEST_PG_DATABASE=<string>
SENTY_DSN=<string>
JWT_SECRET=<string>
```
#### Relevant Trello Tickets
- [Ticket 1](some.com)
